lib/src/kbasesearchengine/main/IndexerWorker.java

```diff
@@ -403,35 +407,40 @@ private EventHandler getEventHandler(final String storageCode)
 
     public void processOneEvent(final StatusEvent ev)
             throws IndexingException, InterruptedException, RetriableIndexingException {
+
+        // update event to reflect the latest state of object for which the specified StatusEvent is an event for
+        EventHandler handler = getEventHandler(ev.getStorageCode());
+        StatusEvent updatedEvent = handler.updateEvent(ev);
+
         try {
-            switch (ev.getEventType()) {
+            switch (updatedEvent.getEventType()) {
             case NEW_VERSION:
-                GUID pguid = ev.toGUID();
+                GUID pguid = updatedEvent.toGUID();
                 boolean indexed = indexingStorage.checkParentGuidsExist(new LinkedHashSet<>(
                         Arrays.asList(pguid))).get(pguid);
                 if (indexed) {
                     logger.logInfo("[Indexer]   skipping " + pguid +
                             " creation (already indexed)");
                     // TODO: we should fix public access for all sub-objects too (maybe already works. Anyway, ensure all subobjects are set correctly as well as the parent)
-                    if (ev.isPublic().get()) {
+                    if (updatedEvent.isPublic().get()) {
                         publish(pguid);
                     } else {
                         unpublish(pguid);
                     }
                 } else {
-                    indexObject(pguid, ev.getStorageObjectType().get(), ev.getTimestamp(),
-                            ev.isPublic().get(), null, new LinkedList<>());
+                    indexObject(pguid, updatedEvent.getStorageObjectType().get(), updatedEvent.getTimestamp(),
+                            updatedEvent.isPublic().get(), null, new LinkedList<>());
                 }
                 break;
             // currently unused
-//            case DELETED:
-//                unshare(ev.toGUID(), ev.getAccessGroupId().get());
-//                break;
+            case DELETED:
```

Comment:
Why not DELETE_ALL_VERSIONS? Unshare has to do with data palettes, not deletion (and I'll grant you, the name and the whole data palettes situation is really confusing)